subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc

```diff
@@ -702,6 +702,55 @@ tasks.create("myTask") {
 // === `Property`, `Provider` and `NamedDomainObjectProvider`
 
 
+[[kotdsl:assignment]]
+== Kotlin Property assignment
+
+Gradle's Kotlin DSL supports Property assignment as an experimental feature that can be enabled by adding `systemProp.org.gradle.unsafe.kotlin.assignment=true` to the `gradle.properties` file.
+Property assignment reduces the verbosity for Kotlin DSL when <<lazy_configuration#lazy_properties,lazy properties>> are used.
+Property assignment works for properties that are _final_ and have type _Property_ or _ConfigurableFileCollection_.
+Property assignment should be preferred over custom property setters since it makes Kotlin DSL code style consistent.
```

Comment:
:x: I feel we can't make this statement right now, given that the feature is still experimental.

```suggestion
Once the feature is released out of experimental status, using the `=` operator will be the preferred way over calling `set()` in Kotlin DSL.
```